enterprise/j2ee.sun.ddui/src/org/netbeans/modules/j2ee/sun/ddloaders/SunDescriptorDataObject.java

```diff
@@ -107,7 +108,7 @@ protected int getXMLMultiViewIndex() {
     private DDType descriptorType;
     private final int xmlIndex;
     
-    public SunDescriptorDataObject(FileObject pf, SunDescriptorDataLoader loader) throws DataObjectExistsException {
+    public SunDescriptorDataObject(FileObject pf, MultiFileLoader loader) throws DataObjectExistsException {
```

Comment:
The more specific type caused reflection errors when instantiated from OpenIDE's `MimeFactory`.

Reply:
Failure in [sigtest check](https://travis-ci.org/github/apache/netbeans/jobs/728712948):
```
Class org.netbeans.modules.j2ee.sun.ddloaders.SunDescriptorDataObject

  "E5.4 - Removing constructor" : constructor public org.netbeans.modules.j2ee.sun.ddloaders.SunDescriptorDataObject.init(org.openide.filesystems.FileObject,org.netbeans.modules.j2ee.sun.ddloaders.SunDescriptorDataLoader)
```

Keep both constructors, if possible. Alternatively let's investigate why this is broken at all: what is the actual failure with the old constructor signature?

Reply:
For the records, the error was:
```
    [junit] java.lang.NoSuchMethodException: org.netbeans.modules.payara.eecommon.dd.loader.PayaraDescriptorDataObject.<init>(org.openide.filesystems.FileObj
ect, org.openide.loaders.MultiFileLoader)
    [junit]     at java.lang.Class.getConstructor0(Class.java:3082)
    [junit]     at java.lang.Class.getConstructor(Class.java:1825)
    [junit]     at org.openide.loaders.MimeFactory.<init>(MimeFactory.java:54)
    [junit] Caused: java.lang.IllegalStateException: org.netbeans.modules.payara.eecommon.dd.loader.PayaraDescriptorDataObject.<init>(org.openide.filesystems
.FileObject, org.openide.loaders.MultiFileLoader)
    [junit]     at org.openide.loaders.MimeFactory.<init>(MimeFactory.java:57)
    [junit]     at org.openide.loaders.MimeFactory.layer(MimeFactory.java:77)
    [junit]     at org.openide.loaders.DataLoaderPool.factory(DataLoaderPool.java:641)
    [junit] Caused: java.lang.reflect.InvocationTargetException
    [junit]     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [junit]     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    [junit]     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [junit]     at java.lang.reflect.Method.invoke(Method.java:498)
    [junit]     at org.netbeans.core.startup.layers.BinaryFS$AttrImpl$MethodAndParams.invoke(BinaryFS.java:730)
    [junit] [catch] at org.netbeans.core.startup.layers.BinaryFS$AttrImpl.getValue(BinaryFS.java:660)
    [junit]     at org.netbeans.core.startup.layers.BinaryFS$BFSBase.getAttribute(BinaryFS.java:506)
    [junit]     at org.openide.filesystems.MultiFileObject.getAttribute(MultiFileObject.java:918)
    [junit]     at org.openide.filesystems.MultiFileObject.getAttribute(MultiFileObject.java:851)
```

Reply:
`MimeFactory` looks specifically for a constructor with `MultiFileLoader` in the signature; maybe it could search constructors for MFL subclass ?

Reply:
Strangely enough, the `@DataObject.Registration` on  `PayaraDescriptorDataObject` could have never worked, IMHO: `MimeFactory` passes in an instance created by `DataLoaderPool.getDefaultFileLoader()`, which is definitely not a SunDDwhatever. Maybe I could save the API change entirely by removing the extra (not effectively used) registration completely, leaving the registration on `PayaraDescriptorDataLoader` only.

Reply:
I have reverted the SunDescriptorDataObject, and removed the extra (and non-working) registration in Payara. Filed [NETBEANS-4828](https://issues.apache.org/jira/browse/NETBEANS-4828).